### ISSUE TITLE
Plugin SDKs: no way to assert what a fix produces (Python and TypeScript)

### ISSUE THREAD

**Issue by walf443** [walf443/nginx-lint#378]
Follow-up from PR #376. Splitting this out so it can land on its own.

## Problem

`PluginTestRunner` (Python and TypeScript) can assert the number of findings, their line, and their message — but nothing about the `Fix` a plugin attaches. A plugin author cannot write "given this config, `--fix` should produce that config".

This is not hypothetical. The guest-side `insert_before` / `insert_before_many` constructors emitted a `Fix` shape that the applier normalizes into a **whole-line replacement**, so an "insert before this directive" fix silently *deleted* the directive from the user's config. It shipped, and unit tests stayed green throughout, because no test could look at applied output. It was caught only by code review, and the same latent bug is still live in the TypeScript SDK (#377).

`insert_after` had a quieter variant of the same problem: the guest emitted unindented column-1 insertions where the host indents each line to the directive's column. Again invisible to the current assertions.

The Python constructors were fixed in #376 and pinned with tests that compare the emitted `Fix` **records** field-by-field against the host's implementations in `src/plugin/component_rule.rs`. That is a decent proxy, but it tests the intermediate representation rather than the outcome, and it only holds as long as someone keeps the two in sync by hand.

## Proposal

Expose fix application to the SDKs so tests can assert the fixed text.

`crates/nginx-lint-common/src/linter.rs` already has the public entry points the host uses:

```rust
pub fn apply_fixes_to_content(content: &str, fixes: &[&Fix]) -> (String, usize)
pub fn apply_fixes_to_content_detailed(content: &str, fixes: &[&Fix]) -> FixApplyResult
```

For Python, `plugins/python/nginx-lint-plugin` (the pyo3 crate) would take a `nginx-lint-common` path dependency alongside its existing `nginx-lint-parser` one, convert the WIT `fix` records into `nginx_lint_common::Fix`, and expose something like `apply_fixes_json(content, fixes_json) -> str`. `PluginTestRunner` then grows:

```python
runner.assert_fixed(
    "http {\n    server_tokens on;\n}",
    "http {\n    server_tokens off;\n}",
)
```

Because it routes through the same applier the CLI uses, a guest-constructed fix that normalizes into the wrong operation fails the test instead of reaching a user's config.

TypeScript needs the equivalent; its testing entry already instantiates the parser as a WASM component, so the applier could ride along the same way.

## Notes

- The conversion has to preserve the range-vs-line distinction exactly (`start_offset`/`end_offset` present means a range fix; otherwise `normalize_line_fix` decides, which is where the original bug came from).
- Worth covering the multi-fix case, since the applier resolves overlapping and same-position fixes in a defined order.
- Once this exists, the host-parity tests added in #376 (`plugins/python/nginx-lint-plugin/tests/test_config_builder.py`) can stay as-is, but the outcome-level tests become the primary guard.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01Vswikio3DwgzxDpbAFrd4S

**Comment by walf443** [comment]
The Python half is done in #385: `nginx_lint_plugin.testing.apply_fixes()` and `PluginTestRunner.assert_fixed()` run a rule's fixes through `nginx_lint_common::apply_fixes_to_content_detailed`, and `Fix` gained `Deserialize` so the JSON crossing the boundary lands in the struct the applier consumes — which removes the line-vs-range conversion this issue warned about, since there is no conversion left. Verified byte-identical to `nginx-lint --fix --rule-only` on the same input.

Leaving this open for TypeScript, which **cannot** be done the same way.

The Python SDK is a native extension, so exposing the applier was just a path dependency on `nginx-lint-common`. The TypeScript SDK reaches the parser as a WASM component, and the obvious move — adding an `apply-fixes` export to the existing `parser` world in `wit/nginx-lint-plugin.wit` — creates a dependency cycle: the applier lives in `nginx-lint-common`, and `nginx-lint-common` already depends on `nginx-lint-parser` (`crates/nginx-lint-common/Cargo.toml:11`), so the parser crate cannot depend back on it.

So TS needs a second component, built from a crate that depends on `nginx-lint-common` and exports the applier, shipped alongside `wasm/parser/` and instantiated by the testing entry the same way. Either that, or a single combined "testing" component that exports both `parse-config` and `apply-fixes` and replaces `wasm/parser/` — the first leaves the existing component untouched, which seems preferable.

Worth doing together with #377, since the applier is what would demonstrate that the corrected TS fix constructors actually behave — exactly how it was used in #385.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01Vswikio3DwgzxDpbAFrd4S
